File: frontend/src/js/child-template.js
import {PolymerElement,html} from '@polymer/polymer/polymer-element.js';

class ChildTemplate extends PolymerElement {
    static get template() {
        return html`<button on-click="handleClick">Child Template</button>`;
    }

    static get is() { return 'child-template' }
}

customElements.define(ChildTemplate.is, ChildTemplate);

File: src/main/java/org/vaadin/example/SubTemplate/ChildTemplate.java
package org.vaadin.example.SubTemplate;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.polymertemplate.EventHandler;
import com.vaadin.flow.component.polymertemplate.PolymerTemplate;
import com.vaadin.flow.router.Route;
import com.vaadin.flow.templatemodel.TemplateModel;

@Tag("child-template")
@JsModule("./src/js/child-template.js")
@Route("child")
public  class ChildTemplate extends PolymerTemplate<TemplateModel> {

    /**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	@EventHandler
    private void handleClick() {
        System.out.println("Click on Button in the child template");
    }
	
	public  ChildTemplate() {
		
	
		
	}
}